### ISSUE TITLE
Welcome new contributors

### ISSUE THREAD

**Issue by bortek** [bortek/EZ-WifiBroadcast#155]
(This message is addressed to people who are in the list of Assignee, not to everyone else)

Hi guys!

EZwfb project that I am hosting code of on Github is in need of new enthusiasts who love the project and are willing to contribute and continue developing it. Thank you "careyer" for bringin this topic up. 

Since you guys have been active on the forum and in the Issues discussions on Git I though you might be one of those enthusiasts :) Therefore I have added you as contributors on Github project which means you can push/pull/merge etc the code. I am sure we all have skills that can benefit the project in a good way. If for some reason you do not wish to be a contributor you can withdraw yourself trough the settings page. 

If you have not worked with Github before please read-up about it or watch YT videos on the ways of working with github. Those who have previously worked with it can hopefully help others to get on track. My suggestion for the structure is the following:

Master branch is locked from changes directly and is supposed to be updated via Pull Requests. 

There is now the new branch called "develop" which is a default branch where active coding is ongoing. Every time someone makes code changes , it is commited and then pushed to develop branch. This can be done several times before creating PR. When you feel ready create a PR towards master branch, discuss the changes in the PR by commenting. Then someone else other than the PR creator have to approve it. Afterwards it can be merged to master. 

Usually after one or several PR when we feel ready for a new version we create a release from the master in Github and add new tag to it. All this is to encourage discussion. Feel free to create Issues and discuss there. It is easy and fun to reffer to the code changes from the Issues. Github is state of the art collaboration tool for dev projects. Sounds complicated? :) Dispair not, it is actually easy when you have done it few times. 

Feel free to contact me by commenting here or via bortek666 (at) gmail dot com if you have any questions, thoughts or suggestions.

Regards
/Boris

**Comment by careyer** [comment]
Thank you Boris for the confidence in us. We will just do our best to serve the project well because we believe in it. Feels awesome to belong to the official enthusiast circle! :-)

**Comment by Yes21** [comment]
Yes Thank you to welcome us into this enthusiasts circle !
I will contribute to update the wiki pages.

**Comment by careyer** [comment]
@Yes21 : Feel free to take a glimpse at my structure suggestion at: https://www.rcgroups.com/forums/showpost.php?p=40335211&postcount=6150. Since I am very experienced with WBC-RC I would volunteer to contribute to the "RC Control Setup" section ;-)

**Comment by htcohio** [comment]
Boris,

Love the project, it has unlimited potential. I'm grateful that you brought me on and I am looking forward to contributing in any way I can.  

I would highly recommend inviting User1321.  He does not engage on the group very much but we have been working together over the last few months to develop new enhancements and optimizations.  And, as a matter of fact today he just received some much needed RPI Hardware he didn't have on hand until now so that should accelerate the progress he was already making.

Cheers,

Justin


ase has already

**Comment by careyer** [comment]
Just overhauled the Wiki section with a new Navigation menu.
Just minor changes ;-)

Please be aware that the pages follow some naming convention for better maintainability. e.g.
`Software : Advanced : Page name` ... try to stick with that convention please.

Now we can go ahead and skim through the pages and update and add information.
For all entries shown in black color e.g. "Software Setups - Quickstart Guide" a new page needs to be added with the "New Page" Button on the top.

**Comment by bortek** [comment]
@careyer Looks good. We try to preserver as much relevant info as possible. 
@htcohio user1321 is already invited, he hasn't accepted it yet. You are still to accept your collaborator invitation.  ;)

Should we keep the image and link to v1.0 ? We dont even have release notes for it and I am sure no one is using it. The others can we keep for historical reasons and to see progress.

**Comment by careyer** [comment]
@bortek : Thank you very much. Yes I have not removed any of the information at all yet. Just restructred things and put things together that belong together. Next step will be to review the information since it is outdated in parts and add additinal infos and topics.

I guess we should keep the link to v1.0 because of "historical" reasons! v1.0 started it all :-)

Someone should also take the effort and update the code in the repository to reflect the latest release candidate 1.6rc6. The code currently shown is as of 1.5 or even older. That scares off a lot of people who want to tinker with the code or contribute based on the most recent version. @bortek : Maybe you can take care of this part? I am not so experienced in the Code section yet.

**Comment by htcohio** [comment]
@Bortek,

Would you mind telling me how to accept the invitation? I'm not seeing a
link or reference here?

**Comment by bortek** [comment]
sure, let's keep v1.0.

Main issue with the code is that we do not have a mechanism to build the image using the source. Normally you have some build platform which sucks in the code from som  repo (Git) and build the image using some base image plus the git code. 

I do not know how rodi did it. I am just guessing here but I think he made the changes inside the the running image via shell and then after many changes the image was released as vX.Y.

Now the latest code is inside the the and I am not sure where to start. It feel a lot of work to just go into the running image and extracting pieces of code and files.  Any suggestions ?

**Comment by careyer** [comment]
@bortek: I see! ... Yes there is an easy way how to extract all the latest source files from the image. Do you happen to know the program WinSCP? You can easiliy copy files between Windows and the running Linux image. 
After all it is just a matter of drag-n-drop. :-)
This would allow to update all the files in the code repository to at least reflect the latest version. I think this is a good start. Many people have asked for it already. 

It might also be a good idea to explain why we use this rather strange "non build mechanism supported" process. It might also be a good idea to invite the user "RespawnDespair" - he forked the latest repo and has setup such an build process. See: https://github.com/RespawnDespair/wifibroadcast-image-builder

@htcohio : You should have received an invitation email from the GitHub repo. Check your mail inbox and spam folder maybe?

**Comment by htcohio** [comment]
> @htcohio user1321 is already invited, he hasn't accepted it yet. You are still to accept your collaborator invitation. ;)

Ok, I got it. You should have received confirmation. I was able to accept through the email link. :)

**Comment by htcohio** [comment]
> 
> @htcohio : You should have received an invitation email from the GitHub repo. Check your mail inbox and spam folder maybe?

I found it, I originally clicked on the link on RCgroups, not the email.

I've been reading and quickly starting to learn how Github works, the terminology etc... I also see that I can directly edit page info for the Wiki and install guide now that I accepted the invite.

I am starting to see how Github is a great tool for keeping collaboration on a project organized; assuming the basic guide is followed properly.

***What do you guys think about possibly changing the project name? Specifically, something that incorporates "FPV or HD" into the title? i.e. OpenHD-Wifibroadcast, FPV-Wifibroadcast or something similar? 

The UAV/FPV community is small by some measurements, my thinking is that the title could provide better insight to what the project is about? Thoughts?

**Comment by Yes21** [comment]
> It might also be a good idea to invite the user "RespawnDespair" - he forked the latest repo and has setup such an build process. See: https://github.com/RespawnDespair/wifibroadcast-image-builder

Yes it would be great if "RespawnDespair" could join us, and also "Lelik" ...

**Comment by htcohio** [comment]
+1 on Lelik because I have a verified image working with the rpi3b+ which makes the ground Wireless 5G hotspot connection easy as cake and works perfectly with 2.4 radios 😉

**Comment by bortek** [comment]
I can check with RespawnDespair

We re not changing the project name at the moment :) . The usage of EzWfb is much more wider than FPV and multirotors. This is a video link system that you can hook up to anything. People have send a sond to the stratosphere with EzWfb and some even wanted to hook it up with some Cubesat.

@careyer I am quit familiar with both Win and Linux tools ;) Any idea which directory from the image to copy ? We can't take the whole root dir as it will include all the lib and binaries, at least they are presnet on any linux system. Or do you mean there is a specific dir we should copy ?

**Comment by matiaspl** [comment]
@svpcom 's fork of the original wifibroacast got some well thought of changes in terms of encoded stream processing, maybe he would backport his changes to this repo too

**Comment by careyer** [comment]
@bortek : Great! Just wasn't sure if you knew WinSCP. Yes we should copy the whole /root and its subdirectories. All the magic happens there. also not forget the .profile which is the most significant file of all that is hidden inside /root

As for the naming of the project: I also think we should keep the name! It is very well established and bortek is absolutrely right with his arguments. However it would be awesome to create a logo or somthing. Does anybody know a graphics designer by any chance? I think a logo is a crucial thing for for professional external effectiveness.

**Comment by RespawnDespair** [comment]
@svpcom did a wonderful job in regards to the actual video transmission. I have a test planned whether or not bidirectional telemetry communication would work with his solution as well. Since this is a major component of EzWFB right now, we need to keep that working. If we can include an even more robust video tranmission based on @svpcom 's work all derivatives should also benefit (e.g. Dronebridge).

In regards to image creation *sigh* i had this working mostly, but instead of just using the exisiting source i split everything up in order to make it more maintainable. I *could* give it another stab, it's in my repository now, but i would like to put it here, since this is the official place. I will be very clear upfront this time. It DOES work and CAN generate useful images, but i need HELP. Collaboration is key here, that's why i'm happy with this step and it reinvigorated my spirits, but i'm no Linux pro and my testing/dev time is limited.

**Comment by pilotnbr1** [comment]
@bortek the files will be quite spread out. The present code and dir in the github gives a starting point. Most will be under the wifibroadcast dir but some will be in the root shell such as the .profile and there are probably some bash scripts that are not even uploaded to github.
 It’s all pretty bastardized between what was Befis original radio tap stuff that you would make, rangarid osd, Rodizio’s start scripts and rc stuff, and a smattering of Arduino...
I might be able to help organize the code depending on time demands.

Edit just saw careyer post regarding just grabbing the whole root. That would be the most comprehensive solution, but I think someone would have to add an explanation as to where all the working parts are so that a new dev doesn’t get lost in all the other raspian crap.

I too think the present name is good as it has been around for a few years and is present in at least three different forums and languages. Logo is a great idea- it will give it more credibility. People get scared off because ezwbc seems so hackish.

**Comment by htcohio** [comment]
Since this is early on in the process,  I just wanted to mention that I've been following this project called SRT by Haivision (secure reliable transport). The technical side is way over my head but it is extremely active everyday and it appears to have some advantages that may be able to be adapted to this project at some point in the future.

I just wanted to bring it to your attention, has anyone else seen this? 

https://github.com/Haivision/srt

**Comment by RespawnDespair** [comment]
> I might be able to help organize the code depending on time demands.

In my repository i took a stab at organizing the code. I also rewrote some things and split the bash script. Unfortunately the bash script i made is broken. I tried to be smarter than i was in regards to loading the configuration script. 

I'm not saying my implementation is the gold standard, but it should provide a starting point, and it works with my image generation...

**Comment by careyer** [comment]
My friends, I'm really glad to see that life is coming back to the project and we have the chance to bring together all the smart ideas and the marvelous work of skillful people for good! I am happy to have kicked this off. You guys are awesome!

**Comment by htcohio** [comment]
> As for the naming of the project: I also think we should keep the name! It is very well established and bortek is absolutrely right with his arguments. However it would be awesome to create a logo or somthing. Does anybody know a graphics designer by any chance? I think a logo is a crucial thing for for professional external effectiveness.
>

@careyer my brother-in-law is a graphics designer. I can definitely get that taken care of this weekend most likely.  You can probably put together a couple Renditions for us to take a look at.  Did you have any ideas or examples of similar logos that you had in mind?

**Comment by RespawnDespair** [comment]
@careyer I feel the same, i have always believed in this project. It has the potential to surpass anything commercially available right now. Perhaps one day we will look back to this as 'the beginning'. 
If i have some spare time this weekend i will draw up a schematic in regards to the structure and branching of the repository for you all to approve. It should simplify building and maintaining the images.

**Comment by pilotnbr1** [comment]
@respawndespair that would be a huge help!

@htcohio thats an interesting project, maybe a better way to handle fec as currently its pretty good but does cost a slight bit of latency.

I see alot of ideas here. I think theres alot of low hanging fruit in terms of documentation, putting the current code on here, and better marketing of ezwbc. In keeping with the github framework and preventing this thread from getting outrageous, perhaps actual enhancements to ezwbc would best be done by submitting a enhancement request and then they can individually be discussed in their own threads.

**Comment by user1321** [comment]
@RespawnDespair 
> @svpcom did a wonderful job in regards to the actual video transmission. I have a test planned whether or not bidirectional telemetry communication would work with his solution as well. Since this is a major component of EzWFB right now, we need to keep that working. If we can include an even more robust video tranmission based on @svpcom 's work all derivatives should also benefit (e.g. Dronebridge).

Good Day\Night\Morning to all!

Believe me, it will work. Also that thing very useful to transfer IP camera (RTSP) and other common messages.
You can see implementation here:
https://www.youtube.com/watch?v=sevEihOZncQ&t=95s
Bidirectional telemetry and IpCamera based at svpcom  code.

**Comment by careyer** [comment]
@htcohio : No I have nothing in mind as for a logo. It should just depict some of the core components like "wireless", "HD", "Streaming" maybe. I just got the idea because the Dronebridge Project looks so much more professional just because of the logo alone. Just let your designer depicts some drafts maybe that we can agree upon here.

@RespawnDespair / @all : Once the schematic and new code structure has  been agreed upon I would like to contribute a working prototype image that user dino.de and I have works on over the course of several month. It offers some major improvements mainly in regards to the RC portion:
-  Mavlink v2.3 support (up to 24Ch RC, solves a lot of the current v1 problems), 
- RC switchable AirPi-GPIOs (enables physical actions on the aircraft)
- RC switchable Ground Station actions. (e.g. switch OSD on/off, change between screens)
- Also my FLIRone Thermal Camera support is included. (Predator Style night vision and switching between visible/thermal video feed ;-)

Maybe we can port some of this good stuff (all tested and working) to the new build.
BTW: dino.de would also be an extraordinary contributor. He donated all the RC stuff in the first place.

**Comment by RespawnDespair** [comment]
@user1321 YES! Having you on board will propel the project even further. I am amazed by the things you have realized. If we can get everything into the repository here that would be one of the biggest steps. I'll get to work on my proposal and then we can collaborate on merging everything!

**Comment by matiaspl** [comment]
@htcohio @pilotnbr1 regarding SRT - from what I know it's not easily adaptable because it needs bidirectional IP communication to work. wbc's FEC has it's own advantages since we use airwaves. One thing we could transplant from e.g. DVB to make the streaming perform in a more reliable way is the Viterbi algorithm. From occusync we could borrow intra-frame refresh with client-side startup NALUs injection for lower latency and error recovery.

**Comment by Yes21** [comment]
**@RespawnDespair** I'm very happy that you join us. I did tested your building script, and was waiting a lot of it. You did an enormous job, and I've been frustrated that it came not to the end. I really think your script will be the best way to generate next releases of EZ-WBC.
**@ALL** Congratulations for this revival :))

**Comment by Yes21** [comment]
**@careyer** It  would also be great to integrate rtl8812/14au drivers. Did you contact "Lelik" ? 
Its last image is working great with AWUS036AC/H and AWUS1900 (but there's a bug in bidirectional telemetry). It would be better if his image could be merged with legacy EZ-WBC ...

**Comment by user1321** [comment]
Generating images with help of Image-Builder probably will be best option. IMHO.
In case if we will have a lot of source codes....
Btw, do Image-Builder generate image that can work at RPi 3 B+?

As for RussianImage... I can see that he use Kernel 4.9.80-v7, but what kind of patches did he apply to  it, some one know?
+ Several times at that image I got Kernel Panic with connected Ethernet, probably RPi 3b+ underpowered.

**Comment by careyer** [comment]
@Yes21 : No not yet... I do not speak russian at all. Maybe @user1321 can contact him? user1321 you are from Ukraine as far as I know, right? Do you happen to speak russian?

**Comment by user1321** [comment]
Also, I want to share information about progress that was achieved working together with Htcohio 
1. Camera Switch that based at "ArduCAM Multi-Camera Adapter Board for Raspberry Pi"
It can switch up to 4 RPi camera`s via RC joystick. ( At one moment of a time only one camera active)
Link to device:
http://www.arducam.com/multi-camera-adapter-module-raspberry-pi/

2. Android application that allow to change most of wifibroadcast-1.txt and  osdconfig.txt (only #define IMPERIAL true, #define COPTER) settings.
Currently it work in next way:
2.1. Phone establish connection with ground RPi  via USB or WIFI.
2.2. Ground RPi  send to phone settings from wifibroadcast-1.txt file.
2.3. In case of settings change at phone:
  Phone send request to ground RPi  to change something in wifibroadcast-1.txt
  Ground RPi  forward request to Air RPi to change something in wifibroadcast-1.txt.
  Ground RPi and Air RPi replay to phone that settings was changed.
I have to say, that at my local side it works normally, but at side of Htcohio it`s work not very stable. So, I have to find and fix problem...

3. Live CD iso file that may be written to USB flash drive and plugged to laptop.
That will start Ununtu with patched kernel that will allow using WN722N receive video from RPi air + receive\send MavLink messages.

At this moment I`ll trying to create image that will boot at RPi 3B+, As soon I will do that, I will try to add to it Android application support and camera switch and will share with you.

P.S. Hope you will be able to decrypt my interpretation of English language :))

**Comment by careyer** [comment]
@user1321 your english is fine! You do not need to worry! May I please ask you to reach out to Lelik (the russion guy who did the rtl8812/14au drivers addition? Thank you very much! Maybe he has a GitHub account also and can contribute?

**Comment by htcohio** [comment]
I reached out to Lelik, He should respond quickly, I've talked to him a
couple times in the last few days.  Username LELIK does not appear to be on GitHub so perhaps his username is different?

@User1321,

I'm telling you your English is absolutely flawless.  I don't know why you think you're typed English is not good.  If I didn't know you I would have just assumed your some guy living in the US or some other English-speaking country. 

A few weeks ago I signed up for this Rcdesign.ru Forum after having trouble
signing up and not getting verification emails for some reason.  FYI if you
sign up the verification emails went into my junk email box.

Also, the translator on this site seems to come up with some interesting
language translations that are nowhere close to what you're trying to say.

Rcdesign forum
http://forum.rcdesign.ru/f90/thread515083-22.html

Lelik's page
http://forum.rcdesign.ru/member.php?u=16248

**Comment by user1321** [comment]
Sure I can contact to him, but firstly I want to ask All you guys, what exactly should I ask?
Personally I interested in what kind of patches did he apply to kernel.
Maybe something more?

___
Add:
And maybe question why he decide to use exactly 4.9.80-v7. Was it random or some reason.

**Comment by htcohio** [comment]
Just to clarify, the camera switch, synchronized settings and,
bi-directional telemetry all work perfectly fine if you use separate UHF or
RC.

Everything seems to freeze up after a minute or so if wfb RC is used.

**Comment by careyer** [comment]
@user1321 : yes please ask him what patches he applied to the kernel to make the  rtl8812/14au drivers work and maybe if he has a GitHub Account (account name) if he wants to contribute to "the original thing"

@htcohio : Sounds like a buffer overflow problem of some kind. Reliability of RC / bidirectional telemetry is cruicial. If we have to decide beweteen rtl8812/14au drivers  and telemetry / rc stability we should go for the later one. As far as I can tell the new MavLink v2 from dino.de works flawlessly and absolutely stable.
Good news btw: Chief devs @ ArduPilot have finally taged my feature request for >8Ch Mavlink-RC https://github.com/ArduPilot/ardupilot/issues/9472 YeahhH!!
btw: @bortek could you please also invite the user dino.de? He is a wizard when it comes to teh RC part of EZ-Wifiboradcast and recently implemented state of the art Mavlink support in a test build

**Comment by Yes21** [comment]
> Sure I can contact to him, but firstly I want to ask All you guys, what exactly should I ask?
Personally I interested in what kind of patches did he apply to kernel.
Maybe something more?

**@user1321**
It would be interesting to know :

- what is the latest branch of aircrack-ng he used (5.2.20 or 5.3.4)

- what are the modifications he did (I think he only did modify .profile and tx_telemetry.c)

- if he will be ok to share these modifications with us

- if he can confirm that bidirectional telemetry is not working

Thanks a lot

Ps : I've looked one time at the russian forum, but google translation seemed me like chinese !!

**Comment by bortek** [comment]
Great discussion and great spirit! :)

@RespawnDespair Your help will be much appreciated. We need to to establish some sort of build boilerplate so we can build an image using Github source (and perhaps raspbian base image). 

Don't have very much time right now but what I plan to start with is to get the code out of an image. Afterwards we can together decide which one we want to merge/update with the existing codebase. If someone else wants to do it please let me know so we don't do a double work :-/

I think we are almost done discussing in this issue. So let each of us open new issue when we want to discuss specific issues. 

Also we should create roadmap of what features do we want to develop/include in the coming releases (say 1 year ahead). Features that other alike projects do not have ;) Just have it in the back of your mind for now.

**Comment by careyer** [comment]
@bortek : Thank you! I am happy to hear what you just said. I was worring that you might think: "OMG what the hell have I done letting them into my Git - They're turning everything to the left!" 
I .. and i guess I speak for the others as well - we are really some hardcore enthusiats. We are sorry for that  :metal:

Thank you for trusting in us!

**Comment by htcohio** [comment]
@Bortek,

It sounds like all the new developers are trying to figure out exactly where everything stands to get our bearings together to determine what needs to be worked on and addressed.  There are several variants floating around out there between all of us and an issue or enhancement request might already be fixed  on a different version that is already verified. For example,  I have several feature request but it sounds like Careyer is or already has completed work on it so there isn't much sense adding that at the moment.

Baby it would be helpful if I started to write up a "road map" similar to what Wolfgang typed up for dronebridge.  Does everyone think that is a good idea? We can add to an edit it as needed.

**Comment by careyer** [comment]
Good point @htcohio - Thank you for pointing this out! Indeed I have a much advanced 1.6rc6 image at hand that is proven to work 100% flawlessly. It enhances the 1.6rc6 image by the groundbraking  features mentioned [here ](https://github.com/bortek/EZ-WifiBroadcast/issues/155#issuecomment-429266470). Because this unofficial "realease" makes some major leaps forward (eg. Mavlink v2 support and Air GPIO control - both crucial for a future AIO system) - and btw solving all the current bi-directional telemetry issues -  I would like to vote to take this as a start and integrate all the other stuff from there.
 
If we really want to build a All-in-One (AIO) system Mavlink v2 support is absolutely crucial for success. For anybody not aware of what can be done with this check out this: [31.2km record flight with WBC-RC](https://www.rcgroups.com/forums/showpost.php?p=39931332&postcount=5116) (credits due to khomyakk)

After all we must agree upon a starting point (anker) from where to start working and integrating. I think the changes done by @user1321 can rather easily (sorry to anticipate this - I have no idea) be applied to this anker and @RespawnDespair provides the structure that will help to get this into a development workflow like it is supposed to be.

If anybody could tell me a destination that I can upload 1,2GB to I would be more than happy to share that anker with you. Everybody okay with that? Any objections?

**Comment by bortek** [comment]
It sounds like you have a good image to start with. Can you upload it to your Google drive and share a link?

**Comment by Yes21** [comment]
> If anybody could tell me a destination that I can upload 1,2GB to I would be more than happy to share that anker with you. Everybody okay with that? Any objections?

:+1: +1

Ps : I suggest this new "anker" could be named : 2.0 rc1 (?!)

**Comment by careyer** [comment]
@All: No Google Drive... Dropbox instead, sorry: https://www.dropbox.com/s/yzavlwsfrcbys57/EZ-Wifibroadcast-1.6RC6_FlirOneG2_v0.9_omxh264_MavLinkv2_GPIO.img?dl=0

Based on a vanilla 1.6rc6 this image integrates: 
-  **Mavlink v2.3 support** (as of 1.6rc6: legancy Mavlink v1 / 8Ch only)
(24Ch RC channel uplink: 18Ch input to Mavlink RC_CHANNELS_OVERRIDE, 6 Ch spare),
- **Switchable GPIOs at AirPi (High/Low) via RC** 
(enables physical actions on the aircraft, e.g. driving relays/lights/landing gears/you name it ...)
-  **RC switchable Ground Station actions.** 
(e.g. switch OSD on/off, change between OSD screens, Toggle video inputs)
-  **Bonus: "FLIRone" Thermal Camera (Gen2) / "insta 360 Air" support** 
(Predator Style night vision and hot switching between visible/thermal video feed ;-) (see #71 )

"FLIRone" in action:
<img src="http://www.ip-networx.de/content/GitHub/EZ_FLIR_01.JPG"> 

Video: https://youtu.be/5aDpvRuO9cI (video was recorded with 1.6rc5)

"insta 360 Air" in action: (thanks to johnboiles)
<img src="http://www.ip-networx.de/content/GitHub/insta360.JPG">

**Some notes:**
- currently ArduPilot is only supporting 8Ch for MavLink RC_CHANNELS_OVERRIDE (but the are working on 18CH support, [see here](https://github.com/ArduPilot/ardupilot/issues/9472): THIS WILL BE HUGE)
- MavLink v2 solves all the current incompatability issues (e.g. "wirelss" setup like compass/ACC calibartion/not all GCS commands supported etc...)
- HD-Raspi Cam & FLIRone thermal video are transfered in parallel (bandwidth for FLIR is negligible: only 160x120px @ 8fps H.264 hardware encoded anyway)
- For your convience & testing purposes only: FLIRone video stream has been replaced by GStreamer Testimage "videotestsrc" (for those who do not own the hartdware). 
**Beware this eats up 100% CPU at AirPi! 
RELAX: Once the GStreamer TestImage is deactiveted CPU load drops to normal**)
- HD-Video/Test-Video can be toggled via Ch16
- Ch15 and Ch24 as of now drive GPIO18 and GPIO 17

To track the changes you might want to search for the keywords: "FLIR", "switches", "airswitches" in the .profile. Feel free to reach out to me with any questions.

**Comment by user1321** [comment]
Guys, I want to ask your opinion...
We all have some kind of ISO images... As for me, I got mine by doing modification to working system via SSH terminal and after changes was done I just clone SD card to iso file.
But, that is not an option anymore as we agreed store our modifications at GitHub.
If we also agreed to use ImageBuilder as base for future releases I see it in next way:

1. Upload "own" code to GitTree.
2. Apply patch to copy of original ImageBuilder to let ImageBuilder  build correct image that will reflect changes that was uploaded at step "1".

Maybe We can talk about it?
Or any other ideas how can we do it?
I just breaking  my head now how we will add code and not move farther and farther from each other in our modification.

**Comment by bortek** [comment]
Of course we should talk about it! We probably everybody has images that they made modification to. But lets agree to select one image as a base and that should the one which has the most number of changes. As far as I can tell it is @careyer image. That has to be stripped down and extracted as files @user1321 do you know what changes you made, like you have some files saved somewhere perhaps ? If not then we will have to strip it down too into files. I can do it but it will take some time. I have an idea to mount IMG file into linux box as a partition and then just copy the files over.

Afterwards these changes need to be merged manually. @user1321 Can you elaborate a little bit on what is "copy of original ImageBuilder" ?

**Comment by user1321** [comment]
>>I have an idea to mount IMG file into linux box as a partition and then just copy the files over.


If mainly you use Windows OS, you may easily do it with "ImDisk Virtual Disk Driver"
Link to Wiki: https://ru.wikipedia.org/wiki/ImDisk

>>Can you elaborate a little bit on what is "copy of original ImageBuilder" ?

Sure, I mean at gitHub there is file that related to "wifibroadcast-image-builder" Create by RespawnDespair:
https://github.com/RespawnDespair/wifibroadcast-image-builder/blob/master/build_images.sh
It contain a numbers of instructions that determinate what  exactly will contain generated ISO image.

If execute image build with all original content it will not reflect changes that can be potentially uploaded to GitTree.

So, Idea is simple:
1. Upload "own\future" source codes to GitTree.
2. Create -> apply patch to build_images.sh
3. Execute patched build_images.sh that will build ISO image that will contain changes from "point 1"

In that case we will have one "stable\start" point and we can create our own modifications based at that "stable" point.

In case if we will patch ISO image`s....
That can be a hell.  Just imagine:
I have a lot of files that adopted to your WFBC 1.6. And suddenly we decide that new image will be (just for example) image that provided by Careyer.
What should I do to adopt it to it? - Very carefully take a  look at every changes that was done with it and apply all patches to it.

It`s ok. 
But what if after several days or months we decide that I have some release candidate?
Do we will copy all changes from Careyer image to mine?

Maybe it`s good reason to use build_images.sh?
All that is just ideas... Maybe i am wrong with it.
That`s why I want to discuss it with All you)
__________________
ADD:
P.S. And that with only two images, what we will do with 3, 4?
_________________
ADD2.
Sorry that I add comments so often, but I was thinking about one more thing.
In case if we will work based at Image..
To apply patches from one release to other we have to documented each line that was changed or use any tool that can compare files line by line... That will not be very easy task...

**Comment by careyer** [comment]
@user1321 : I totally get your idea which I believe is good and indeed the only smart way to go. Unfortunatly I am not a coder myself and all that code related stuff is a bit difficult for me. (most changes in my image have been made by dino.de - he is also on github btw. <- @bortek ;-) ). 
His changes are profound and almost essiantial for the future sucess and reliablity in regard to a AIO system. They pretty much stabilize the foundation of bi-directional telemetry and RC control and take it to the next level.  This is why I suggested taking this as "anker"-image.

Please don't get me wrong: I really like your changes (e.g. configuring via FPV_VR) a lot and I am looking forward that we will integrate them in a common image. But unlike things that stabilize the foundation and reliability/robustness of the image a  GUI based configuration feels somewhat a cosmetic thing to me. Again: Don't get me wrong here. It is just all about to make a smart decission from where to start.

@user1321 maybe you could kindly try to dismantle my image and try to apply the 3 steps you just proposed above? Then we would have a good starting point. I am happy to assist you in identifying the changes made to .profile and explain about the v4l2loopback kernel module stuff needed for the Flir/insta360 integration (I have documented this in a step by step guide). All the other stuff (rcswitches, airswitches, mavlink2) should be pretty much file based (c Executables)
Would this be a way to go?

**Comment by htcohio** [comment]
@Careyer,

I'm sorry but I have to 100% disagree that that synchronized configuration
settings are cosmetic because, when most people go to the field with
multiple aircraft they may go with a long-range fpv plane running 3s, a 6S
drone and a 4S plane.  Also, it's much easier to work up a instructional
guide and video when there's a graphical user interface with selectable
options as opposed to text files.

Battery cell count, aircraft type and desired video range settings
ultimately need to be easily changed in the field.  Having the ability to
interface with an app is by far the easiest and most logical way of going
about it.  I know from experience because I've taken all these things to
the field and fiddled for hours trying to SSH or modify the SD cards and it
really takes away from the enjoyment.

Furthermore, rpi3+ compatibility is a must because of the benefit of the
internal 5g wifi connection.

The usable image situation as of right now is as follows:

@Careyer image functions:
•Flir camera campatable
•GPIO trigger w/ RC
•Mavlink 2
•RC GS osd on/off

@user1321 image functions
•GUI config
•GPIO 4 camera switcher
•PI3 compatibility (working on bugs)
*I Believe user1321 has successfully compiled and tested out
RespawnDespairs image Builder that works with rpi3b+.

Correct me if I'm wrong but once those bugs are figured out isn't that the
lattice that should be used to adapt the other functions from careyer and
user1321????

So.... In my humble opinion I think there's 3 priorities as of this moment.

1.)  Both of yours have GPIO trigger (those should be able to be merged in
addition to the camera switcher which only needs 2 things; 1. PWM values
for (1, 2, 3 or 4 cameras).  Also, define channel # for switch.

2.)  Mavlink 2.3 (this is a must)

3.) Rpi3b+ compatibility


-- Once that is working and stable and fully integrated with the
respawndespair image Builder then, work can be done to integrate the other
bells and whistles.

-Thermal camera support
-RC ground rc/osd control
-RTL8812/4 5g compatibility (I'll never use it but I know many people want
it but I've tested and it's kind of buggy)


Ultimately, there's a lot of ideas out there but in order to establish a
solid foundation for the future, getting the image Builder working is
critical.

I think I covered everything here but let me know if I missed anyting.

**Comment by Yes21** [comment]
> Guys, I want to ask your opinion...
We all have some kind of ISO images... As for me, I got mine by doing modification to working system via SSH terminal and after changes was done I just clone SD card to iso file.
But, that is not an option anymore as we agreed store our modifications at GitHub.
If we also agreed to use ImageBuilder as base for future releases I see it in next way:

**@user1321** : I've seen you made today big progresses in making **RespawnDespair's** ImageBuilder script to work better. We thank you very much for your efforts !

- When it will be confirmed that this script is able to build a perfectly working EZ-WBC 1.6rc6 image, I suggest that this image could become the starting point of new developments. It will be a "new 1.6rc6" (or 1.6rc7, or 2.0rc1, or ... ?). This image wouldn't have any new features, except a new kernel and support for RPI3B+. 
- The code used to generate this "first" image (the one a few restructured by **RespawnDespair**) should be hosted on this github and become the reference code for future developments.
- After that, each developer of new features (since 1.6rc6) could submit its code to a dev branch on this github.
- Everyone should be able to test the new features by building itself a test image with the ImageBuilder script.
- When the new features are well tested and stable, we could generate and publish a new public image = new version. And the dev branch could become then the master branch.
- And so on ...

**@ALL** : What do you think of this proposition ?

**Comment by htcohio** [comment]
Yes, that's probably the best way.  So basically an equivalent of 1.6 RC6 that is compatible with pi3b+... If possible, I think it should also include the Mavlink v2 compatibility as well.

2.0 would be a good version.  I'm sure this would quickly result in a couple beta (test) versions with the other features previously listed with the intent of eventually getting to ezwfb V2.1




We can all test it out.

**Comment by careyer** [comment]
Yes this is pretty much perfect!
First we need to be able to generate a working 1.6rc6 with @RespawnDespair ImageBuilder. From there we can add feature by feature from the various images arround.

**Comment by user1321** [comment]
Hi Guys!
I what to ask you to help find bugs inside image attached in this post.
That image is image that generated by ImageBuilder + it contains all fixes that listed at:
https://github.com/RespawnDespair/wifibroadcast-image-builder/issues/10

Reason why I asking you to help is - to find out what else must be fixed before we can use it.
I almost sure that it may contain some more bugs that must be fixed before we can use it.
So, please don`t fly at it! Seriously! Just ground!!! And better to unattach from motors. Just to be in safe!
When we will find other bugs and fix them also, we can integrate appended solution list and integrate them to Build script.
I think it`s good place to report about bugs at RespawnDespair GitHub page.
I will be checking it from time to time...
P.S. I checked it under RPi 3B+ and RPi 2.
P.S. It`s starts up a little longer than original WFBC 1.6. Don`t be surprised.
I think some optimization to Linux OS is required. But, first lets find and fix bugs.

Link:
https://drive.google.com/open?id=1JJpdEo1xrH97nNVpmZqp_nxmXmPFJWCP

__________________________________
ADD:

Just minute ago found another bug.
If ground RPi not connected with ethernet (RPi3, RPi2) OSD not appears.
I will find out why...Just connect Ethernet if you have to Ground RPi.
Bug 11.
solution:
8.5 inside osd_rx_functions.sh return back:
/tmp/osd >> /wbc_tmp/telemetrydowntmp.txt &
ps -ef | nice grep "/tmp/osd" | nice grep -v grep | awk '{print $2}' | xargs kill -9

ADD2:
Just in case image with fixed Ethernet dependency:
https://drive.google.com/open?id=1bG4b0y8Q5hfiWhDISG8zS9IRHxfiI3MI

**Comment by RespawnDespair** [comment]
Aside from actually getting the image generator working (many thanks to @user1321 for finding all of the problems) i also promised a suggested git structure.

People who have looked at my repository know i have split everything to different repositories, this is both good and bad, it makes it easier to work on a separate piece, but you loose the overview. My main reason for doing this was the fragmentation that was already happening, i wanted to be able to change the image builder to git clone another repository and then use that.

Since we are now pooling resources i want to suggest the following:

```
project root
│   README.md
│
└───base
│   
└───osd
│   
└───rc
│   
└───status
│   
└───scripts
│   
└───misc
│   
└───video
│   
└───image-builder
```

So everything under one repository, separated in folders. This may seem counter-intuitive, but it makes sense since we can then create branches of all the relevant code that makes up EZ-WFB. In such a branche we would prepare the next release image. When done we make a pull-request to master. In this way, we make sure master always represents the latest released image while keeping an overview of all code that makes up the system.

Off course suggestions or improvements are welcome, once i have the image-builder working with @user1321 's changes i will make this structure in my own account. We can then test it and move it to @bortek 's repository once validated. That way we will prevent further annoyance/confusion.

**Comment by careyer** [comment]
@RespawnDespair  - Nice! Maybe add a dedicated "telemetry" directory as well and rename "hello-video" to just "video" (some AddOns might also use make use of GStreamer, ffmpeg and such)

**Comment by RespawnDespair** [comment]
You are correct on the video folder, i will change that. As for a telemetry folder it would make a lot of sense, but the telemetry code is mixed with other code right now (at least in the last image i checked). If that's still the case we could keep it in misc or base and extract it to a separate folder in the next release...?

**Comment by careyer** [comment]
sure. You may take a look at the image i have linked above (the MavLink2 version) and investigate if it is still mixed there with the basecode. I can't tell because I did not implement it myself. All that magic was done by dino.de - the current telemetry based on Mavlink v1 must be treated as legacy code causing unneeded trouble.

**Comment by RespawnDespair** [comment]
I'll try to get all of @user1321 's findings fixed in the separated repos. Then i will merge to one repo. Then we'll do one final sweep over all the different images to make sure we included all the intended functionality! *so much work 😄*

**Comment by careyer** [comment]
You are doing a great job - don't worry!

**Comment by user1321** [comment]
@RespawnDespair , please, don't hurry , I think there are more things to fix. 
I will say when it's time )
P.s I think its better to set reg zone to US, not to BO or CN

**Comment by RespawnDespair** [comment]
@user1321 i'm not in a hurry, let's make sure everything works before moving on to the next step. When i implement your regfix i will set it to US instead of BO.

**Comment by careyer** [comment]
@All: I created a "Help maintain the Wiki Tasklist" on Google - feel free to jump in for a free slot: https://docs.google.com/spreadsheets/d/1o-zk3RvxrfGni4EYF-C5TQYLIx2TO9t_FtJvEVsTzzU/edit?usp=sharing

**Comment by htcohio** [comment]
@careyer,

I definitely have some notes organized for the wiki list.  Before I publish anything I'm hoping to actually field test some of the updated recommendations for the wiki based on the newest image that they are working on.  There are just certain aspects that are unclear and I think need a second look and field test such as the SR settings, antenna separation and usb cable length.  

My plan is to have a couple craft I'll fly the same Mission with different configurations with antennas and little modifications to see how they compare.

**Comment by user1321** [comment]
Guys. I just checked mavlink-forwarder program from Image Builder and it generated some error during start:
ionice -c 3 nice /home/pi/mavlink-router/mavlink-routerd -e $192.168.2.2:14550 /dev/pts/1:57600

Error:
Could not open conf file '/etc/mavlink-router/main.conf' (No such file or directory)
Could not set DTR/RTS (Invalid argument)

Probably it can be easly fix, But if I will take a look at it in any case, maybe we replace it with  Careyer solution with Mavlink2? (I did`t look at it yet.)
Or will stay with default one?
How do you All think?

**Comment by htcohio** [comment]
Guys,
Hey, we have $193 in donations so far!!!

Once we hit $400 I'm raffling off a Ubiquiti Wi-Fi station EXT 😉.

We'll have to figure out how it is split ultimately.

As far as I know, user1321 and respawndespair are the only ones primarily working on the updated version correct?

These donations are intended to be applied to all work that started on Friday 10/12. 

Could you guys possibly try to track actual hours worked on the code?  This way it could be split accordingly.  there's no easy way to do this because many people have put many hours into the project but considering the amount of work that's required to decode and establish a solid foundation I'm thinking this should help keep things moving in the right direction.

Great work!

**Comment by careyer** [comment]
@user1321 : MavLink v1 is legacy and likely  to cause more errors in the ongoing work. There is no point in spending time with something that is already end of life and also currently causing a lot of problems and hiccups during operation. We should definitely go with Mav2

**Comment by htcohio** [comment]
@careyer,

I'm just curious, is V2 actually stable in the image that other guy you were working with developed?

I thought you said otherwise, or is it that the 18ch rc is not yet fixed by the mavlink developers? 

Also,  as for the donations that are coming in rather quickly, I forget what developer you were working with on some of those enhancements.  Perhaps you could help me find out or discover who is doing what from Friday October 12th onward?

**I just don't want a programmer/developer that has put significant time into something being excluded from the total amount brought in from contributors but at the same time, I don't want the "pot" to be diluted too much.  

Perhaps the top 2-3 Developers for each stage...

Stage 1:. Complete image Builder with MavlinkV2 compatibility as the base for future development..
(User1321/Respawndespair/ 3rd???)

Stage 2: integrate the newest enhancements previously mentioned.  
(Tbd....../tbd...)

Stage 3: bug fixes and continued development.
(Tbd...)

*I don't want to over-complicate things nor do I want anyone to feel unfairly excluded from the "active developer" list...

**Comment by bortek** [comment]
I have now extracted files (including .profile) from careyer's 1.6RC6 image as well as from official 1.6RC6 image and put it up in two separate branches called respectively. Please take a look. Any idea which files should be updated. It is not possible to merge alltogether since files on the img are residing differently than in repo.

**Comment by htcohio** [comment]
@Bortek,

As of this moment my understanding is User1321 and respawndespair are finalizing the image Builder and I was able to test it yesterday and I think we're about 80% there but I'm not the coder perhaps you should refer to this thread for further information and updates.
I also believe user1321 is looking at mavlinkv2 integration from careyer/dino.de image.
https://github.com/RespawnDespair/wifibroadcast-image-builder/issues/10

**Comment by Yes21** [comment]
> Probably it can be easly fix, But if I will take a look at it in any case, maybe we replace it with Careyer solution with Mavlink2? (I did`t look at it yet.)
Or will stay with default one?
How do you All think?

We did decided that the first image created with ImageBuilder should be exactly the same as the legacy 1.6rc6 (except new kernel and Pi3+ support).
But if you have too many problems to make Mav1 work again, I can understand that it could be worth to try to replace it right now by Mav2 from Careyer/Dino.
Wouldn't it be a too much changes for this first image ? 
There are so many things to fix before the first ImageBuilder image works perfectly like the last 1.6rc6 !
I don't know the best solution ...

**Comment by careyer** [comment]
@htcohio : 

>> I'm just curious, is V2 actually stable in the image that other guy you were working with developed?
>> I thought you said otherwise, or is it that the 18ch rc is not yet fixed by the mavlink developers?

There are two sides of the MavLink v2 implemenation: EZ-WBC (Raspi) and ArduPilot (Fligthcontrol). EZ-WBC implematation by dino.de for EZ-WBC works aboslutely flawless and (besides the 24Ch uplink) gives several improvements alongside plain RC-control. Just to name a few:  V1 had several problems due to missing message types - e.g. a.) some Mission Planner messages caused the telemetry link to crash completely or b.) the drone just refused to do what the user was commanding it to do - e.g. flight mode changes sent from Mission Planner were just ignored or c.) functions like Compass calibration did not work at all due to missing Ardupilot specific command extensions. Ardupilot uses V2 now per default but is not yet supporting the 18Ch channel extension introduced with Mav 2.3. They are working on it right now. (also RCgroups user davidgp is taking care of it right now)
To cut a long story short: We can either be ahead of Ardupilot (without any drawbacks and immediate benefits) or we can stick with old Mav1 (legacy for more than 2 years now) and will get rendered useless within a matter of month. If you have not yet familiarized with the differences of the protocol you should do so.

>>Also, as for the donations that are coming in rather quickly, I forget what developer you were working >>with on some of those enhancements. Perhaps you could help me find out or discover who is doing >> what from Friday October 12th onward?

>>**I just don't want a programmer/developer that has put significant time into something being >>excluded from the total amount brought in from contributors but at the same time, I don't want the >>"pot" to be diluted too much.

dino.de and me, we do not want to have any of the donated money at all. You and me, we have talked about this idea before and I believe that I clearly stated that I belived it was not a good idea to initate a donation/reward system. This might work short term but I have seen many projects fail because people got argueing about the equitable distribution of money. I believe this is a hobby project and everyone contributes with the best of energy, time, skill and commitment that he can. I do not believe that this is a question of hours spent or plain skill level but more about passion. And sadly there is no scale for that. You should know that. However this is my very personal view of things. Your opinion and that of others may differ - I am okay with that.
But to answer your question in terms of man-hours: dino.de and the team put several month of work and testing into what I have released to the wild above. We are follwoing the project from the very beginning (page 1 on RCgroups so to say) and have seen all the UPs and Downs. Trust the word of a man almost being to old to play with this kind of toys.

Due to a personal tragedy in the closest circle of friends that i got to know just 2 days ago I have decided to reduce my activities here for the time being -  Best of luck guys!

Cheers!

**Comment by htcohio** [comment]
I'm really sorry to hear that man.  Go take care of your business.  I have you and your friends in my thoughts.

As for the donations, there were several people that expressed interest in this and we're at a point where quite a bit of time and effort is needed to establish a new starting point.  This may only be a one-time thing just to accelerate the process since there's so many great enhancements that can be integrated.  This is why the intention of the funding is to be directed towards all work that was started on October 12th onward.

**Comment by careyer** [comment]
Thank you for your compassion! Just one last thought... this is a hobby project. It has stalled for many month .. almost a year. I am happy to have kicked this off but think about it this way: We are not so much in a hurry. It pretty much does not matter if we have that "new build" one, two or three weeks from now.  In any case, we will achieve 100 times more than in the past months. Take your time and do it the correct way instead of rushing things. Drive it with passion not money and it will be a long term success story.

**Comment by user1321** [comment]
Very sorry to hear. Wish you all the best.

**Comment by Yes21** [comment]
**@careyer** : I can understand what you say "Trust the word of a man almost being to old to play with this kind of toys."
I'm also a soon old man, and know that the life is sometimes making these toys/hobbies not so important.
I thank you a lot to have given a new energy to this project.
But now take care of you, and of your family and friends.
You will be welcome here, when you want.
"Bis bald"

Yves

**Comment by bortek** [comment]
I concur with careyer’s though about donation thing. For me this has been a hobby project on my free time and I will continue to contribute in the same way without needing any donation.

**Comment by RespawnDespair** [comment]
@careyer take care of yourself, your family and your friends. We'll be here when you feel like you're ready again.

As for being paid to do work on the project i feel the same as most others. This is for me a hobby project, introducing money muddies the waters and will never be 'fair' to all involved. I think the best thing we could do is keep the money centrally and spend it on hardware and/or hosting.
Now if anybody does feel like getting paid for their work i won't object, that's a personal choice.
@user1321 spends more time on the project than i am able to due to work and family life. His efforts moved the project forward a lot. The same can be said for dino.de 's efforts. 

So for me, money is not necessary, now if anybody has a larger wing left over that i can rebuild with EZWFB, that's a different matter (Crashed my ZOHD Orbit...)

**Comment by pilotnbr1** [comment]
I think one addition to the wiki could also be a common question page. I would be willing to write that up (or at least the questions). Hopefully it would alleviate some of the repetitive question asking on the rcgroups thread. As much as people get encouraged to use the thread search function, few ever do.

@careyer best wishes as you take care of you and yours. Hope to see you come back here renewed. Nice work on the google drive doc assigning wiki pages! Will you still be writing up the rc stuff?

**Comment by user1321** [comment]
Next image:
It contains bug fixes: mavlink-routerd, gst-launch, hotspot, DHCP.
Using that image I able to connect via HotSpot to Ground RPi via phone, get parameters vis Ground Сontrol app (Android), receive video over raw h264 and RTSP.

As for RTSP, it confuse me a little.
Using RTSP I get video with delay something like 0.5 second +

It`s look to me very strange, because I remember that using one of my images (not published here) It was significantly lower.
So, is it normal? ( I am not saying acceptable. Just to know, as we trying to create a copy of original WFBC 1.6)

Also I did same test at "RussianImage" at RPi 3B+ and got same result with build in wifi card (RP 3B+) and with external one.
So, please, check it also. And if you remember or know that it must (!) be lower inside GroundControl app ( I am not talking about VR_2018 app) . Please, say about it.

Also, this is not the end. Just some report about progress.
P.S. Adjust settings.sh file for your configuration.

https://drive.google.com/open?id=1Mg7O1tQmn-GTEdkfCjWZdnZWXAhG58ga

**Comment by careyer** [comment]
@pilotnbr1 :+1:  a FAQ section is always a good idea. What was I thinking forgetting about that? Go ahead add it to the wiki ;-) And yes, I am still considering adding to the RC section. However this has time and I will follow along the project for a while first. Who knows what is going to happen and going to change.

@All: Thank you for your kind words. I do not intend to completely leave the project behind - not yet. To calm you down: Nobody passed away. It is just that due to a sudden and very severe accident/illness of a close friend I cannot focus my thoughts and energy solely to the project right now. I'm sorry about that and I feel a little guilty to Boris that I have to hand over the rest of the coordination now so soon after I've kicked it all off. That's especially because I feel at the very moment that we're running into a situation where not everyone is going to emerge as a winner and that doesn't benefit the project in the long run. We need to find a way to make decisions democratically and not just get started and present participants with faits accomplis. Otherwise we will loose skilled and knowing people one after the other over the course of time. I have to confess that this uncoordinated fundraising campaign in particular annoys me to such an extent that it spoils my desire to participate.  It just doesn't reflect my spirit and mindset. 

@user1321 & @RespawnDespair : :+1: I think you are doing a fantastic job and I strongly believe this is the correct way to go. So the above is nothing against you and your work. Just wanted to express my personal feeling about the team spirit and the lack of a roadmap agreed upon all.

**Comment by user1321** [comment]
Just found one more problem: 13. Some Error (crash ) in case of USB phone connected.
So, please, connect it via WiFi at that moment.

**Comment by pilotnbr1** [comment]
@careyer yes faq is better.
Please dont let the fund raising turn you off. Personally I am against this as well. IMHO money detracts from the pure hobbyist nature of this project (most of us already work and think about money the rest of the day). I think the fund raising is well intentioned and htcohio is just trying to contribute. I think perhaps some cultural differences may be at play, since we are like the united nations here (I am american living in Germany traveling in Helsinki), which is an aspect I enjoy as well. However, I do not think compensation is something anyone on here is looking for. Personal time to play with this seems to be the big constraint (can you gift me time lol). I believe any fund raising at this point should be on the side and removed from development as it only spams these already busy threads. We all have already spent some time during an exciting push, to stop and consider this money issue. I will not say where, when, or how any money should be used because I want no responsibility over it. Maybe at some point down the road there will be some financial need, but it is not now.

@careyer I hear you with how uncoordinated development/actions can make this frusterating. Early days of wbc was chaos with scripts being shared and everyone running different images. Rodizio solved it by being a one man band but in the end I think things did not go the way he hoped or expected. Mostly in regards to dronebridge imho. In the end with these open, loose colaborations they can go in directions that nobody predicted or wanted. Github could provide more structure I think. Most important is to all stay together and keep developing. I just hope everyone will continue to be inspired even if things go sideways a little. :)

Edit and sorry if i sound so negative! I am not- I am very encouraged and excited by this latest push! Wbc is a great project and very relevant and useful to many ppl

**Comment by Yes21** [comment]
> I feel a little guilty to Boris that I have to hand over the rest of the coordination now so soon after I've kicked it all off.

> We need to find a way to make decisions democratically and not just get started and present participants with faits accomplis.

I can make a proposition to help giving this project a more democratic coordination : 

I propose to create a dedicated issue for a proposition of a **roadmap**. I could write a first draft in trying to gather propositions of the various developers, and to order them on a timeline (without dates or duration).
Then each developer could express himself and give his opinion about the draft roamap.
I will update the roadmap project untill everybody is ok (if that's possible ...).

(_I don't want to coordinate the project because I'm not a developer, but just an engaged user. When I'm using an open source project, I like to try to get it better, especially when the project is awesome.
I'm 53 years old, and in a past life I've been an engineer and project manager. I'm a linux user since 15 years, and had a little contributed to a few open source projects for a time (only translations into french, guides or website update).
Now I'm working alone, and just begin to try to earn my life with forest cartography, with drones and multispectral analysis (Parrot Sequoia camera)._)

**I will write this draft of a roadmap only if each implicated developer agree with my proposition.**

ps : like the others, I'm not interested by money ...

**Comment by user1321** [comment]
@pilotnbr1
money detracts from the pure hobbyist nature of this project ...

Brr....
It`s good when some one talk own opinion. I think we have good word for that - IMHO
But look, starting from Friday I spent at that project something around 25 hours  of work (maybe more, I didn`t take a lot attention at time). I don`t mean when you drink coffee and looking to windows it`s a work, I mean direct 25+ working hours.

Basically it`s look like at: from morning till deep night with coffee break and pause to do some homework, shopping...
And taking that in attention we still don`t have working image.

And now imagine like it can be in normal way. For example 1-2 hours every day after work.
So, in that way we will reach same result in half month, maybe more.

That donation it's not normal level of salary even for "Cleaning Man"
But why you think I or someone else can`t be rewarded with that kind of promotion?

**Comment by bortek** [comment]
@yes21 I dicussed with careyer about making a roadmap couple of days ago and that we need to have it. So yes I think it is vital to have one so we can prioritise and decide. If you will do so please do it on Trello. It's a tool just for this purpose where one creates a board with cards and ca discus and which can be linked to Github.

**Comment by bortek** [comment]
@user1321 I am sorry that it turned out like that to you. But keep in mind that there is no pressure and no one forces us to do what we do and we have no deadlines to deliver. So let's enjoy what we do in a pace that suits each of us and more importantly have fun when we do it. 😊

**Comment by pilotnbr1** [comment]
Started FAQ page. Feel free to jump in edit/correct/complete!

**Comment by htcohio** [comment]
@All,

Look, please disregard all of this, it's my fault and it was not my intention to create any issues.  I have an entire 6 months of experience under my belt with open source projects and I also own a small business so, in my world if I'd like to see things done quickly paying money is usually the best way to make it happen.  Furthermore, several people had inquired over the last few months about how to donate to the project, I saw it as an opportunity that could only bring benefit and ironically had quite the opposite effect from what I intended.

I have refunded all of the public donations with a note that they can feel free to individually donate money or Hardware privately as they see fit.  

I apologize again, hopefully we can just drop this whole thing and focus on what needs done. 

Justin

**Comment by RespawnDespair** [comment]
@htcohio No harm no foul, everybody has the best intentions as did you.
Open source is probably the only place people will get upset if you try to give them money 😄 

Now as you said, we'll keep doing what we where doing!

**Comment by user1321** [comment]
https://donate.mozilla.org/en-US/?test=no-img
https://donate.wikimedia.org/wiki/Ways_to_Give
https://ubuntubudgie.org/contribute
https://www.archlinux.org/donate/
https://www.debian.org/donations
U sure?
and so on..
There a lot more...

**Comment by bortek** [comment]
Thanks Justin, appreciate it! It was rightly done. 

Now after this long and somewhat productive discussion, where we got to know each other a little bit better, shell we get going? @RespawnDespair I agree and yes please keep going.

@user1321 we are far from the projects you mention above and if someone is expecting to be compensated for the working hours then I think this is the wrong place to be.

Now before we close this Issue can we agree on the proposal to draft the  roadmap as @Yes21  suggested? Just say yes if you agree and if not please come with the suggestion.

**Comment by careyer** [comment]
Yes

Von meinem iPhone gesendet

**Comment by RespawnDespair** [comment]
Yes

**Comment by htcohio** [comment]
@all,

I test it out this image with rpi3b+ ground, pi0w air.

No apparent issues with Telemetry or RC.  Video latency on the rtsp
qgroundcontrol app seemed fine, the video was on a big TV and the latency
has always been higher on this TV compared to my regular computer monitor.

USB tethering appeared to work fine as well. It did not display the "USB
device detected" message however.

2312-25xx mhz appears to be working with full power.

I'll continue to test and update you.

https://youtu.be/Idk2Y4J0eDk


Edit:
2nd video test with pi2 and pi3b+ simultaneously ground/ pi0 air
https://youtu.be/pD5k5h0G8SI

**Comment by pilotnbr1** [comment]
👍

**Comment by bortek** [comment]
Ping @pilotnbr1 @htcohio @user1321 

"Now before we close this Issue can we agree on the proposal to draft the roadmap as @Yes21 suggested? Just say yes if you agree and if not please come with the suggestion."

**Comment by htcohio** [comment]
Yes

**Comment by pilotnbr1** [comment]
Yes

**Comment by bortek** [comment]
@user1321 please let us know if you are still with us, is interested to continue to contribute and if you agree with above.

**Comment by Yes21** [comment]
**@bortek**

> I dicussed with careyer about making a roadmap couple of days ago and that we need to have it. So yes I think it is vital to have one so we can prioritise and decide. If you will do so please do it on Trello. It's a tool just for this purpose where one creates a board with cards and ca discus and which can be linked to Github.

I don't know Trello now, and don't to take time to learn it now. So I will create a dedicaced issue.

I'm busy finishing my first draft.

But @user1321 didn't already give his opinion about my proposition ...

**Comment by user1321** [comment]
yes

**Comment by Yes21** [comment]
**@htcohio** can you remind me in a few words what are the new features you have in your image, and which could be integrated in a future ez-wbc image ?
Thanks.

**Comment by careyer** [comment]
@Yes21 : I think you can find the features in his message [here](https://github.com/bortek/EZ-WifiBroadcast/issues/155#issuecomment-429565132). GUI config is with a modified version of the FPV_VR App I believe. 4-way Camera switcher needs additional hardware - believe this [one](https://www.robotshop.com/de/de/arducam-multi-kamera-adaptermodul-fur-den-raspberry-pi.html?gclid=EAIaIQobChMIgYqY_-ON3gIVwed3Ch0jGQDpEAQYAyABEgLld_D_BwE). Is this correct @htcohio 

@All 
Sidenote: The support for RPi3B+ is a double-edged sword -The higher computing power of RPi3B+ is not really necessary (it is currently only 5% on average for a regular RPi3B). As far as I know, Rodizio has made massive adjustments to the current kernel to optimize performance and newer kernels had stability issues (not sure though if this was a objective observation). We have to be careful not to lose these optimizations when switching to a newer kernel. On the other hand, RPi3B+ support is a must have/fundamental as it significantly lowers the entry threshold for new users who are likely to buy the latest incarnation of the Pi. Also the 5Ghz hotspot option is a big plus. Just some thoughts about this to keep in mind and need to be reflected in the upcomming roadmap.

**Comment by Yes21** [comment]
**@careyer** Thanks for the link.
I agree with your opinion for RPI3B+. It's not so useful (except internal wifi 5Ghz), but as many new users have it, it should be supported ...

**Comment by htcohio** [comment]
@yes21,

Sure, I'm just going to include top to bottom as I understand it right now.

1. The new image Builder with 3b+ and 5g 8812/14au compatibility 
2. Mavlink2 + more rc ch.
3. GPIO trigger camera switcher (probably similar code to @careyers image).
4. COM mechanism which allows synchronized changes to most of the wifibroadcast.txt through an app such as (fpv.vr, qground tower etc...)

**Ideas**
1.) A dedicated gpio and hardware button that changes settings back to a predefined "default" state.  The benefit of this would be if you have to vehicles that were previously with different settings, you would not need to SSH or pull the SD card out to match them.  

2.) Some kind of basic GUI for modifying the layout or components of the OSD.  Perhaps something similar to the minimosd  gui (remember that thing haha)?

**Comment by careyer** [comment]
@All: Another thing that comes to my mind:
Maybe it is usefull to have some sort of compile options/flags that allow to compile or not compile certain AddOns. More and more AddOns/Extensions are to come and it might not be possible to integrate them all to a single image. Some of them might even interfere with or prevent each other.

So the idea would be to have a clean base image first (only the very core features). Then have like checkboxes before the build process to tick all the desired features. e.g:
- [X] EZ-WifiBroadcast Core **(mandatory)**
- [ ] MavLink1
- [X] Mavlink 2
- [X] GUI configuration
- [ ] 4-way ArduCam Switch
- [ ] Switchable GPIOs
- [ ] FLIR
- [X] insta 360Air support
- [X] mAh readout in OSD
- [ ] display: total mAh, km, mins
- [ ] support for rtl8812/14au 
- [X] GroundPi Battery Status

... more to come

Do you get the idea? Is that possible to do in some sort of a automated build pipeline or something like this?

**Comment by dposea** [comment]
Yes from me. 

I am a developer but to be very honest I'm not the guy to lead any project. My interest span and available time are a problem. I can work on a project for hours every day until something new catches my attention then I get obsessed with that. I'm a little odd and my focus level on my current interest is near 100%. I forget to sleep and eat, sit in front of a computer for so long i get cramps, and my girlfriend/roommate wants to kill me because i dont hear her when she talks to me. 

I think the people already interested in updating ezwbc have the necessary skills, and I will help out when I can because I spent literally hundreds of hours figuring out what rodizio had done and how it all worked. His goal was an image anyone could use by changing a few text files, and he met that 1000 times over. An easy to sustain project was not a prority and that is his choice and I will defend that choice any time anyone has something negative to say about ezwbc because of it. 

Anyway, that's who I am and how I think and why someone else needs to take the lead :)

**Comment by careyer** [comment]
> **Ideas**
> 1.) A dedicated gpio and hardware button that changes settings back to a predefined "default" state. The benefit of this would be if you have to vehicles that were previously with different settings, you would not need to SSH or pull the SD card out to match them.

I believe this is already implemented (see [config profiles](https://github.com/bortek/EZ-WifiBroadcast/wiki/Software-:-Advanced-:-Configuration-profiles)) 

> 2.) Some kind of basic GUI for modifying the layout or components of the OSD. Perhaps something similar to the minimosd gui (remember that thing haha)?

Excellent idea! That would be very helpful. Yes the MinimOSD was awesome.... (the OSD was not...hahah)

**Comment by dposea** [comment]
I just read the proposal to get away from images. There are too many add ons to keep going that way, and it makes updates very difficult. 

In regards to the kernal the changes made by rodizio are not extensive and almost all were to add options to wifi drivers or mac80211 headers. Lowest possible latency was by trial and error, picking the best combination of existing programs and tweaking them to get improvements. I dont think there is any way to reduce latency much more without some serious effort and much more control over video processing. That wont happen on an rpi and proprietary video drivers.  Regardless you must have support for the 3+. 

I'll think about how to get past the image distribution. Splitting up the big script is a good start!!

**Comment by bortek** [comment]
Goodie. Then I will close this issue (it has already diverted from the title :) quite a lot)  We already have a roadmap Issue open and you can open new Issue for each task that needs discussion. Good job everybody! 🥇
